lisa/_kmod.py

```diff
@@ -2267,4 +2285,23 @@ def from_target(cls, target, **kwargs):
     def _event_features(cls, events):
         return set(f'event__{event}' for event in events)
 
+    def install(self, kmod_params=None):
+        try:
+            self.uninstall()
+        except Exception:
+            pass
+
+        probe_cmd = 'modprobe {} {} version={}'.format(
```

Comment:
Seeing as the busybox variant doesn't have the option we use, how likely is it that a target device someone tries to run the Lisa module one doesn't have access to modprobe?
If we rely on insmod here we also need to hardcode the path where the module is in the kernel which makes this more error prone. If someone renames the directory the modprobe variant will still work but the insmod variant will break so it's a tradeoff.

Reply:
> Seeing as the busybox variant doesn't have the option we use, how likely is it that a target device someone tries to run the Lisa module on doesn't have access to modprobe?

Any busybox-based system will not have `modprobe -d` available (so any Linux board in our CI), and they may not have modprobe at all. The main problem is not so much the availability of modprobe more than the exact flavor of modprobe. With that PR, we already hit the issue that the system modprobe had `-d` in your setup, but won't in CI mainline linux setups.

We have been bitten by that in devlib multiple times (e.g. the `ps -C` issue). We also had cases of slightly different behaviors (can't remember which command). So since we do have a set version of modprobe + insmod available, we should try to use them.

> If we rely on insmod here we also need to hardcode the path where the module is in the kernel which makes this more error prone.

Isn't that what you did already ? My suggestion is try insmod with your hardcoded android path, then fallback on modprobe if insmod failed. This way we don't have to teach modprobe to look anywhere else. We can speedup the process by running a snippet of shell script:

```
path = quote(path)
busybox = quote(target.busybox)
target.execute(f'([ -e {path} ] && {busybox} insmod {path}/lisa.ko {options}) || {busybox} modprobe lisa {options}', as_root=True) 
```

Reply:
> Any busybox-based system will not have modprobe -d available (so any Linux board in our CI),

Ah yeah that's fair, I was just thinking about phone and normal linux targets, I guess the CI boards do swing the tradeoff the other way.

> Isn't that what you did already ? 

For half the path, this is the full one:
```
/vendor_dlkm/lib/modules/5.10.107-android13-4-00025-g72ae232e302a-dirty/kernel/drivers/soc/arm/vh/kernel/lisa/lisa.ko
```
I meant that modprobe can detect the part from `kernel` onwards, insmod cannot. But I can replicate that with `busybox find` just as well so sure, insmod it is.